### DISCUSSION TITLE
*MAT_MODIFIED_CRUSHABLE_FOAM using LsDyna key file

### DISCUSSION THREAD

**Discussion by ktommy2** [OpenRadioss/OpenRadioss#2372] [Q&A]
LSDyna has a material type 63 that is intended for crushable foam. There is also the option of using a modified version of this, that includes strain rate dependency. I believe this has been available since version R12.
Radioss offers the LAW33 (FOAM_PLAS) material model for foams. As far as I know, it does not consider strain rate dependency. Instead, another material model, LAW70 (FOAM_TAB), has been implemented.

In my LsDyna key file, I have defined an EPS material with *MAT_MODIFIED_CRUSHABLE_FOAM. The simulation runs with LsDyna.
When I try to run it with OpenRadioss, I get an error.

ERROR ID : 100210
** ERROR IN THE INPUT OPTIONS
DESCRIPTION :  
-- BLOCK: *MAT_MODIFIED_CRUSHABLE_FOAM
-- LINE: *MAT_MODIFIED_CRUSHABLE_FOAM
Unrecognized option: *MAT_MODIFIED_CRUSHABLE_FOAM

The error message seems to be clear, but I will ask the question anyway. :)
Is there a conversion error from LsDyna to OpenRadioss or is the function not yet supported?
Thank you in advance for your feedback!

**Comment by ktommy2** [comment]
Although I tried to do some research before asking the question in the forum, as is so often the case, one only finds out what the answer is after the question has been asked...
In this case it is exactly as the error message says: the material model is not implemented.
I found the list of the implemented Commands from LsDyna:
https://2022.help.altair.com/2022/simulation/pdfs/radopen/AltairRadioss_2022_ReferenceGuide.pdf

Another error message suggested to look in the reference manual for the implemented commands.
Note: for new users like me, this sentence would have helped in case of the message about the unrecognized option ;)

**Comment by MarianBulla** [comment]
Hi ktommy2,
and thanks for your interest on OpenRadioss.
Yes, the strain-rates for a crushable foam are not supported, yet.

However, if you want to go for an energy absorbing foam (with Poissons value = 0), including strain rates, can you try to use *MAT_126 (MODIFIED_HONEYCOMB), instead ? It will be mapped to LAW50, which has the strain rates. You just need to use the same curves for all directions, then it will behave also isotropic.

Hope it helps you, as a workaround ?

Thanks and best regards,
Marian

**Comment by ktommy2** [comment]
Thank you and thanks for the amazingly fast support. Also, many thanks for the hint!
Yes, I have done the conversion of the material. First step was to "throw away" everything that could not be converted. So deleted most of the shell control card, etc...
Right know im experimenting with a few material models like MAT_026 (using LCSR), MAT_057 etc... I worked with Abaqus for over a decade, and about a year ago I had to switch to LSDyna. In my spare time I'm trying to learn OpenRadioss. I also want to mention here how happy I am that this explicit solver has become opensource.
In the end, all roads lead to Rome :)

Further comment:
The simulation ran for quite a while, because DT dropped rapidly, then it crashed with the message zero or negative volume. 
Initially i thougt, I messed up the material definition, but than i realized that OpenRadioss uses different units as expected.

I defined the following Unit for LS-Dyna:
*CONTROL_UNITS
         mm         s  mtrc_ton

In the out file i get the following setup:
    UNIT SYSTEMS DEFINITION     
    -----------------------     
                                                          MASS                LENGTH              TIME
    WORK UNIT SYSTEM . . . . . . ( Mg , mm , s   ) 1.0000000000000E+03 1.0000000000000E-03 1.0000000000000E+00
    INPUT UNIT SYSTEM  . . . . . ( Mg , mm , s   ) 1.0000000000000E+03 1.0000000000000E-03 1.0000000000000E+00


After that, i tried to define the following for LS-Dyna:
*CONTROL_UNITS
        mm         s        kg
         1         1     0.001

The units mm, s and kg were succesfully defined, however, the optional control card for unit scaling does not seem to be supported by OpenRadioss.	 
This may be a different topic...

**Reply by MarianBulla** [reply]
Hi @ktommy2,
Thank you very much for your kind words, and I'm happy, you like working with OpenRadioss.

Regarding your point, can you provide your model, please ?
(or a simplified version of it, which is still runable)

I would like to have a deeper look into it and try to see, what to improve.

Thanks and best regards,
Marian

**Reply by ktommy2** [reply]
Hi Marian!
Sorry, it would have made more sense to attach the models in advance...
After a few weeks I can't retrieve everything, but I tried to recreate the simple models. As I said, this is about the principle of dealing with units in OpenRadioss.

Normally I work with mm-s-t and default is m-s-kg in OpenRadioss.

[Radioss_testmodell.zip](https://github.com/OpenRadioss/OpenRadioss/files/14571064/Radioss_testmodell.zip)
In the zip file, there are 3 folders:
mskg_working --> model works as it is
mmst and mmskg001 --> units are messed up.

Thank you in advance for your support!
